Products/Zuul/facades/metricfacade.py

```diff
@@ -620,3 +676,82 @@ def getMetricsForDevices(self, devices, metrics, start=None,
                     results[key][metricnames[metric]].append(dict(timestamp=ts, value=value))
 
         return results
+
+    # This method is supposed to run as a facade method job.
+    def renameDevice(self, oldId, newId, joblog=None):
+        # joblog is injected in the FacadeMethodJob class when this method runs
+        # as a job. The messages written to joblog will be displayed in the Jobs
+        # pane of Zenoss.
+
+        success = False
+
+        try:
+            path = RENAME_URL_PATH
+            request = {'oldId': oldId, 'newId': newId}
```

Comment:
See comments in query. These should probably contain the trailing "/" to prevent renaming the wrong metrics. Also consider breaking up into 2 queries, one to rename metrics and the other to rename tag values.